pkg/steps/source.go

```diff
@@ -129,27 +139,33 @@ func (s *sourceStep) Run(ctx context.Context, dry bool) error {
 		return fmt.Errorf("could not resolve clonerefs source: %v", err)
 	}
 
-	return handleBuild(s.buildClient, createBuild(s.config, s.jobSpec, clonerefsRef, s.resources, s.sshSecretName), dry, s.artifactDir, s.dryLogger)
+	return handleBuild(s.buildClient, createBuild(s.config, s.jobSpec, clonerefsRef, s.resources, s.sshSecretName, s.oauthSecretName), dry, s.artifactDir, s.dryLogger)
+}
+
+func determineCloneURI(sshSecretName, oauthSecretName, org, repo string) string {
```

Comment:
Passing flags as arguments is a smell: https://martinfowler.com/bliki/FlagArgument.html

Reply:
yes this is smelly. I will take a look.

Reply:
Those args are not related to the flags. The flags are pointing to the file that contains the secret.
Those are the names of the generated secret. The names are a hash code related to their contents. 

I guess we need a better way to determine the cloneURI, since we don't initially allow both secrets

Reply:
I am saying that passing an argument to a method to use it in an `if` statement is smelly